File: modeler/src/main/java/net/sayaya/document/modeler/model/ModelToDTO.java
package net.sayaya.document.modeler.model;

import net.sayaya.document.data.Model;
import lombok.experimental.UtilityClass;

@UtilityClass
public class ModelToDTO {
	public Model map(net.sayaya.document.modeler.model.Model entity) {
		return new Model().name(entity.name()).cohesion(entity.cohesion());
	}
}

File: modeler/src/main/kotlin/net/sayaya/document/modeler/model/ModelToDTO.kt
package net.sayaya.document.modeler.model

import org.mapstruct.Mapper
import org.mapstruct.ReportingPolicy

@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.IGNORE)
interface ModelToDTO {
    fun toModel(entity: Model): net.sayaya.document.data.Model
}